R/qenv-get_var.R

```diff
@@ -44,14 +48,28 @@ setMethod("get_var", signature = c("qenv.error", "ANY"), function(object, var) {
 })
 
 #' @rdname get_var
-setMethod("[[", signature = c("qenv", "ANY"), function(x, i) {
-  get_var(x, i)
-})
-
 #' @export
 `[[.qenv.error` <- function(x, i) {
   stop(errorCondition(
     list(message = conditionMessage(x)),
     class = c("validation", "try-error", "simpleError")
   ))
 }
+
+#' @export
+names.qenv.error <- function(x) NULL
+
+#' @export
+`$.qenv.error` <- function(x, name) {
+  # Must allow access of elements in qenv.error object (message, call, trace, ...)
+  # Otherwise, it will enter an infinite recursion with the `conditionMessage(x)` call.
+  if (exists(name, x)) {
+    return(NextMethod("$", x))
+  }
+
+  class(x) <- setdiff(class(x), "qenv.error")
+  stop(errorCondition(
+    list(message = conditionMessage(x)),
```

Comment:
We need to keep some of the classes otherwise, `conditionMessage` fails

```r
Browse[1]> conditionMessage(unclass(x))
#> Error in UseMethod("conditionMessage") : 
#>   no applicable method for 'conditionMessage' applied to an object of class "list"
```